### PR TITLE
Add new CLI command

### PR DESCRIPTION
**Pull request by jinapark202**
This PR adds a new cli command, `./bin/run scaffold`, which takes in a JSON and based on the JSON provided, scaffolds the necessarily files/folders and generates the types.

Schema for JSON in [SDD](https://docs.google.com/document/d/171eaRLeGSrgCT9hNJVS4pQO4Sc6OsbPPD4Q36Exm4bc/edit#heading=h.ki319sb0388)

Click [here](https://docs.google.com/document/d/1vvcD9kBYvJbakYlZ2u_-Uzrj0htMC4VtWLDvD7EjrB4/edit) to access Loom of CLI command.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -96,7 +96,8 @@
       }
     },
     "hooks": {
-      "init": "./src/hooks/init.ts"
+      "init": "./src/hooks/init.ts",
+      "lowcode": "./src/hooks/lowcode.ts"
```
Comment (SyedWasiHaider) [inline packages/cli/package.json:100]: nitpick but should we call it `scaffold`

**Comment by prayansh** [comment]
is there a schema for the json input? would be great if u linked that in the PR as well

File: packages/cli/src/commands/lowcode.ts [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    return integrationDirs
+  }
+
+  parseFlags() {
+    return this.parse(Init)
+  }
+
+  async run() {
+    const { args, flags } = this.parseFlags()
+    const answers = await autoPrompt(flags, [
```
[the remark this answers is not in the document]
Comment (jinapark202) [reply packages/cli/src/commands/lowcode.ts:67]: So right now, the json will be sent to the support team via a Zendesk ticket

File: packages/cli/package.json (hunk 1 above)
Comment (jinapark202) [reply packages/cli/package.json:100]: yeah that sounds good to me :)

**Comment by jinapark202** [comment]
> is there a schema for the json input? would be great if u linked that in the PR as well

The [SDD](https://docs.google.com/document/d/171eaRLeGSrgCT9hNJVS4pQO4Sc6OsbPPD4Q36Exm4bc/edit#heading=h.ki319sb0388) has a rough schema for the JSON input.

File: packages/cli/src/commands/scaffold.ts [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    ])
+
+    const name = answers.json.name
+    const slug = answers.json.slug
+    const authenticationScheme = answers.json.authenticationScheme
+    const actions = answers.json.actions
+    const directory = answers.directory
+
+    // For now, include the slug in the path, but when we support external repos, we'll have to change this
+    const slugWithoutActions = String(slug).replace('actions-', '')
```
Comment (SyedWasiHaider) [inline packages/cli/src/commands/scaffold.ts:83]: Is there a reason we're doing this?

Comment (jinapark202) [reply packages/cli/src/commands/scaffold.ts:83]: We have the same logic for the `init` command.. essentially the folder name for `actions-test` becomes `test`.. will keep it to be consistent

File: packages/cli/src/commands/scaffold.ts [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    }
+
+    try {
+      renderTemplates(
+        snapshotPath,
+        targetDirectory,
+        {
+          destination: slug
+        },
+        true
```
Comment (SyedWasiHaider) [inline packages/cli/src/commands/scaffold.ts:111]: Minor nicpick but it's hard to grok what this `true` param is doing without looking up renderTemplates. Maybe you can define a const above like:

`const overwriteExisting = true`

and then use that..

File: packages/cli/src/commands/scaffold.ts [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+    if (!destination) {
+      return
+    }
+
+    const stats = fs.statSync(file)
+    const parentDir = stats.isDirectory() ? file : path.dirname(file)
+
+    const settings = {
+      ...(destination as BrowserDestinationDefinition).settings,
```
Comment (SyedWasiHaider) [inline packages/cli/src/commands/scaffold.ts:190]: Why is this browserdestination?

**Review by SyedWasiHaider — APPROVED** [review]
Mostly minor comments but LGTM. Looking forward to seeing more iterations on this.

**Pull request merged**